pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSinkBuilder.java

```diff
@@ -0,0 +1,257 @@
+     * @param producerConfig     The configurations for {@code PulsarProducer}
+     * @param connectionSupplier Pulsar client supplier
+     */
+    public PulsarSinkBuilder(
+            @Nonnull String topic,
+            @Nonnull Map<String, Object> producerConfig,
+            @Nonnull SupplierEx<PulsarClient> connectionSupplier
+    ) {
+        checkSerializable(connectionSupplier, "connectionSupplier");
+        checkSerializable(producerConfig, "producerConfig");
+        this.topic = topic;
+        this.producerConfig = producerConfig;
+        this.connectionSupplier = connectionSupplier;
+    }
+
+    /**
+     * @param schemaSupplier Pulsar messaging schema supplier.
+     */
+    public PulsarSinkBuilder<E, M> schemaSupplier(
+            @Nonnull SupplierEx<Schema<M>> schemaSupplier
+    ) {
+        checkSerializable(schemaSupplier, "schemaSupplier");
+        this.schemaSupplier = schemaSupplier;
+        return this;
+    }
+
+    /**
+     * @param extractValueFn extracts the message value from the emitted items.
+     */
+    public PulsarSinkBuilder<E, M> extractValueFn(
+            @Nonnull FunctionEx<? super E, M> extractValueFn
+    ) {
+        checkSerializable(extractValueFn, "extractValueFn");
+        this.extractValueFn = extractValueFn;
+        return this;
+    }
+
+    /**
+     * @param extractKeyFn extracts the message key from the emitted items.
+     */
+    public PulsarSinkBuilder<E, M> extractKeyFn(
+            @Nonnull FunctionEx<? super E, String> extractKeyFn
+    ) {
+        checkSerializable(extractKeyFn, "extractKeyFn");
+        this.extractKeyFn = extractKeyFn;
+        return this;
+    }
+
+    /**
+     * @param extractPropertiesFn extracts the message properties from the emitted items.
+     */
+    public PulsarSinkBuilder<E, M> extractPropertiesFn(
+            @Nonnull FunctionEx<? super E, Map<String, String>> extractPropertiesFn
+    ) {
+        checkSerializable(extractPropertiesFn, "extractPropertiesFn");
+        this.extractPropertiesFn = extractPropertiesFn;
+        return this;
+    }
+
+    /**
+     * @param extractTimestampFn the function that extracts the timestamp from the emitted item.
+     */
+    public PulsarSinkBuilder<E, M> extractTimestampFn(
+            @Nonnull FunctionEx<? super E, Long> extractTimestampFn
+    ) {
+        checkSerializable(extractTimestampFn, "extractTimestampFn");
+        this.extractTimestampFn = extractTimestampFn;
+        return this;
+    }
+
+    /**
+     * See {@link SinkBuilder#preferredLocalParallelism(int)}.
+     */
+    public PulsarSinkBuilder<E, M> preferredLocalParallelism(int preferredLocalParallelism) {
+        this.preferredLocalParallelism = Vertex.checkLocalParallelism(preferredLocalParallelism);
+        return this;
+    }
+
+    /**
+     * Creates and returns the Pulsar {@link Sink} with using builder configurations set before.
+     */
+    public Sink<M> build() {
+        String localTopic = topic;
```

Comment:
I stored them in the local variables to prevent serialization of the whole class. Since the class has no unserializable field and function and this function definition is not transferred so much, I think it is better to change the code to the way that serializes the class.